ludwig/features/audio_feature.py

```diff
@@ -0,0 +1,301 @@
+            audio_feature = np.expand_dims(audio, axis=-1)
+        elif(feature_type in ['stft', 'stft_phase', 'group_delay']):
+            audio_feature = np.transpose(AudioBaseFeature._get_2D_feature(audio, feature_type, audio_feature_dict, sampling_rate_in_hz))
+        else:
+            raise ValueError('{} is not recognized.'.format(feature_type))
+
+        feature_length = audio_feature.shape[0]
+        broadcast_feature_length = min(feature_length, max_length)
+        audio_feature_padded = np.full((max_length, feature_dim), padding_value, dtype=np.float32)
+        audio_feature_padded[:broadcast_feature_length, :] = audio_feature[:max_length, :]
+        return audio_feature_padded
+
+    @staticmethod
+    def _update(audio_stats, audio, sampling_rate_in_hz):
+        audio_length_in_s = audio.shape[-1] / float(sampling_rate_in_hz)
+        audio_stats['count'] += 1
+        mean = (( audio_stats['count'] - 1) * audio_stats['mean'] + audio_length_in_s ) / float(audio_stats['count'])
+        mean = calculate_incr_mean(audio_stats['count'], audio_stats['mean'], audio_length_in_s)
+        var = calculate_incr_var(audio_stats['var'], audio_stats['mean'], mean, audio_length_in_s)
+        audio_stats['mean'] = mean
+        audio_stats['var'] = var
+        audio_stats['max'] = max(audio_stats['max'], audio_length_in_s)
+        audio_stats['min'] = min(audio_stats['min'], audio_length_in_s)
+        if(audio_length_in_s > audio_stats['max_length_in_s']):
+            audio_stats['cropped'] += 1
+
+    @staticmethod
+    def _get_2D_feature(audio, feature_type, audio_feature_dict, sampling_rate_in_hz):
+        window_length_in_s = audio_feature_dict['window_length_in_s']
+        window_shift_in_s = audio_feature_dict['window_shift_in_s']
+        num_fft_points = audio_feature_dict['num_fft_points'] if 'num_fft_points' in audio_feature_dict else get_length_in_samp(window_length_in_s, sampling_rate_in_hz)
+        window_type = audio_feature_dict['window_type'] if 'window_type' in audio_feature_dict else 'hamming'
+
+        if(feature_type == 'stft_phase'):
+            return get_phase_stft_magnitude(audio, sampling_rate_in_hz, window_length_in_s, window_shift_in_s, num_fft_points, window_type)
+        if(feature_type == 'stft'):
+            return get_stft_magnitude(audio, sampling_rate_in_hz, window_length_in_s, window_shift_in_s, num_fft_points, window_type)
+        if(feature_type == 'group_delay'):
+            return get_group_delay(audio, sampling_rate_in_hz, window_length_in_s, window_shift_in_s, num_fft_points, window_type)
+
+    @staticmethod
+    def add_feature_data(
+            feature,
+            dataset_df,
+            data,
+            metadata,
+            preprocessing_parameters
+    ):
+        set_default_value(
+            feature['preprocessing'],
+            'in_memory',
+            preprocessing_parameters['in_memory']
+        )
+
+        if(not 'audio_feature' in preprocessing_parameters):
```

Comment:
I think `if x not in y` is a general convention rather than `if not x in y`. Also no need for parentheses.

Reply:
changed if convention commit fb3abbb